lnrpc/routerrpc/router_backend.go

```diff
@@ -780,3 +781,48 @@ func UnmarshalMPP(reqMPP *lnrpc.MPPRecord) (*record.MPP, error) {
 
 	return record.NewMPP(total, addr), nil
 }
+
+// MarshalHTLCAttempt constructs an RPC HTLCAttempt from the db representation.
+func (r *RouterBackend) MarshalHTLCAttempt(
+	htlc channeldb.HTLCAttempt) (*lnrpc.HTLCAttempt, error) {
+
+	var (
+		status      lnrpc.HTLCAttempt_HTLCStatus
+		resolveTime int64
+	)
+
+	switch {
+	case htlc.Settle != nil:
+		status = lnrpc.HTLCAttempt_SUCCEEDED
+		resolveTime = MarshalTimeNano(htlc.Settle.SettleTime)
```

Comment:
Also use a single `ResolveTime` field on `htlc`?

Reply:
i'm not sure what benefit it offers, we would still have two optional fields: `ResolveTime` and `Preimage`. keeping them isolated i think makes the logic simpler, o/w settle is `ResolveTime != nil && Preimage != nil`, and failed is `ResolveTime != nil && Preimage == nil`

Reply:
Makes sense